scripts/ccpp_datafile.py

```diff
@@ -311,7 +443,29 @@ def _retrieve_dyn_const_routines(table):
 def _find_var_dictionary(table, dict_name=None, dict_type=None):
 ###############################################################################
     """Find and return a var_dictionary named, <dict_name> in <table>.
-    If not found, return None"""
+    If not found, return None
+    # Test valid table with dict_name provided
+    >>> table = ET.fromstring("<ccpp_datatable version='1.0'><var_dictionaries>"\
+                "<var_dictionary name='banana' type='host'><variables>"\
+                "<var name='hi'></var></variables></var_dictionary>"\
+                "<var_dictionary name='orange' type='api'></var_dictionary>"\
+                "</var_dictionaries></ccpp_datatable>")
+    >>> _find_var_dictionary(table, dict_name='orange').get("name")
+    'orange'
+
+    # Test valid table with dict_type provided
+    >>> _find_var_dictionary(table, dict_type='host').get("name")
+    'banana'
+
+    # Test no table found (expect None)
+    >>> _find_var_dictionary(table, dict_name='apple')
```

Comment:
Does this test still work if you explicitly write `None` as the answer?  If so then I might do so just to make it clear that is what will be returned by the function.

Reply:
ooh i learned another thing today - doctests ignore Nones. Added an is None check (& expected result = True)